<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MolPad">
<title>Case Study: Cheese • MolPad</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Case Study: Cheese">
<meta property="og:description" content="MolPad">
<meta property="og:image" content="https://kaiyanm.github.io/MolPad/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MolPad</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/cheese.html">Case Study: Cheese</a>
    <a class="dropdown-item" href="../articles/data_input.html">Data Input</a>
    <a class="dropdown-item" href="../articles/FAQ.html">FAQ</a>
    <a class="dropdown-item" href="../articles/navigate.html">Network Navigation</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Case Study: Cheese</h1>
            
      
      
      <div class="d-none name"><code>cheese.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://kaiyanm.github.io/MolPad/">MolPad</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette provides a comprehensive guide on using MolPad for a
case study, including data pre-processing, network generation, and
result analysis.</p>
</div>
<div class="section level2">
<h2 id="washed-rind-cheese-microbial-communities">Washed-Rind Cheese Microbial Communities<a class="anchor" aria-label="anchor" href="#washed-rind-cheese-microbial-communities"></a>
</h2>
<p>Cheese making is an ancient craft that involves the coagulation of
milk proteins to form curds, which are then separated from the liquid
whey. The curds are processed, shaped, and aged to develop the desired
texture and flavor. Various factors, such as the type of milk, specific
cultures of bacteria, and aging conditions, contribute to the unique
characteristics of each cheese. One crucial step in this process is the
aging practice, where regular washing with a brine solution plays a
significant role.</p>
<p>In the process of producing cheese, regular washing with a brine
solution is an aging practice that can homogenize the microbial
communities on the cheese’ surface and facilitate intermicrobial
interactions. In the following parts, we will analyze a longitudinal
data set of three washed-rind cheese communities collected during cheese
ripening.</p>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p>For the analysis, we use a data set contained in the study by Smith
et al. (2022) on <a href="https://journals.asm.org/doi/10.1128/msystems.00701-22" class="external-link">microbial
communities in cheese</a>. The original research investigated the
successional dynamics that occur within cheese rind microbial
communities using a combination of 16S rRNA amplicon, Illumina, and
PacBio sequencing. We functionally and taxonomically annotate (using
eggNOG (21) and MMseqs2 (22)) the contigs they have generated from the
Illumina reads, to demonstrate the utility of MolPad using a
single-omic. Specifically, we focus on Cheese Sample A and Cheese Sample
C. For more detailed information on our attached data, please check <a href="https://kaiyanm.github.io/MolPad/reference/cheese.html">this
documentation</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"cheese"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">cheese</span><span class="op">)</span></span>
<span><span class="co">#&gt; tibble [106,239 × 18] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ ID     : chr [1:106239] "1" "2" "3" "4" ...</span></span>
<span><span class="co">#&gt;  $ A_1    : int [1:106239] 38 23 24 3 58 12 1 14 1 3 ...</span></span>
<span><span class="co">#&gt;  $ A_2    : int [1:106239] 23 6 5 2 14 9 1 7 1 1 ...</span></span>
<span><span class="co">#&gt;  $ A_3    : int [1:106239] 27 4 37 4 45 14 0 14 3 5 ...</span></span>
<span><span class="co">#&gt;  $ A_4    : int [1:106239] 5 0 10 2 13 4 1 4 4 0 ...</span></span>
<span><span class="co">#&gt;  $ A_5    : int [1:106239] 11 9 19 16 32 13 0 4 1 1 ...</span></span>
<span><span class="co">#&gt;  $ C_1    : int [1:106239] 13 21 3 56 82 2 4 17 7 2 ...</span></span>
<span><span class="co">#&gt;  $ C_3    : int [1:106239] 1 1 0 1 3 0 0 0 0 1 ...</span></span>
<span><span class="co">#&gt;  $ C_4    : int [1:106239] 3 0 1 7 8 1 0 1 2 2 ...</span></span>
<span><span class="co">#&gt;  $ C_5    : int [1:106239] 0 2 4 17 3 4 0 1 8 6 ...</span></span>
<span><span class="co">#&gt;  $ GO_ID  : chr [1:106239] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ KEGG_ID: chr [1:106239] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ domain : chr [1:106239] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ phylum : chr [1:106239] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ class  : chr [1:106239] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ order  : chr [1:106239] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ family : chr [1:106239] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ genus  : chr [1:106239] NA NA NA NA ...</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">annotations</span><span class="op">)</span></span>
<span><span class="co">#&gt; tibble [86,156 × 9] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ ID     : chr [1:86156] "9" "10" "11" "12" ...</span></span>
<span><span class="co">#&gt;  $ GO_ID  : chr [1:86156] "-" "-" "-" "-" ...</span></span>
<span><span class="co">#&gt;  $ KEGG_ID: chr [1:86156] "-" "-" "-" "-" ...</span></span>
<span><span class="co">#&gt;  $ domain : chr [1:86156] "Bacteria" "Bacteria" "Bacteria" "Bacteria" ...</span></span>
<span><span class="co">#&gt;  $ phylum : chr [1:86156] "Pseudomonadota" "Pseudomonadota" "Pseudomonadota" "Pseudomonadota" ...</span></span>
<span><span class="co">#&gt;  $ class  : chr [1:86156] "Alphaproteobacteria" "Alphaproteobacteria" "Alphaproteobacteria" "Alphaproteobacteria" ...</span></span>
<span><span class="co">#&gt;  $ order  : chr [1:86156] "Caulobacterales" "Caulobacterales" "Hyphomicrobiales" "Hyphomicrobiales" ...</span></span>
<span><span class="co">#&gt;  $ family : chr [1:86156] "Caulobacteraceae" "Caulobacteraceae" "Bartonellaceae" "Brucellaceae;-_Brucella/Ochrobactrum group" ...</span></span>
<span><span class="co">#&gt;  $ genus  : chr [1:86156] "Caulobacter" "Caulobacter;-_unclassified Caulobacter" "Bartonella" "Brucella" ...</span></span></code></pre></div>
<div class="section level3">
<h3 id="pre-process">Pre-process<a class="anchor" aria-label="anchor" href="#pre-process"></a>
</h3>
<div class="section level4">
<h4 id="data-and-annotations">Data and Annotations<a class="anchor" aria-label="anchor" href="#data-and-annotations"></a>
</h4>
<p>Here we select a ‘type’ column with <code>phylum</code> to describe
the characteristic of cheese data. Also, the columns <code>phylum</code>
and <code>class</code> are taken as tags for elemental composition.</p>
<p>In this section, we will introduce data preparing steps for analysis.
The annotations dataset contains various columns that describe the
characteristics and properties of samples. First, we select the ‘type’
column with <code>phylum</code> to provide a broad categorization of the
microbial communities present on the cheese surface. This categorization
helps in understanding the overall composition and diversity of microbes
at a high taxonomic level. Then we run the <code><a href="../reference/pre_process.html">pre_process()</a></code>
function to clean and standardize the data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cheesedata</span> <span class="op">&lt;-</span> <span class="va">cheese</span> <span class="op">|&gt;</span> </span>
<span>         <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ID</span>, <span class="va">A_1</span><span class="op">:</span><span class="va">C_5</span>, <span class="va">phylum</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>         <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>type<span class="op">=</span><span class="va">phylum</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>            <span class="fu"><a href="../reference/pre_process.html">pre_process</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>To further annotate the dataset, we also use the columns
<code>phylum</code> and <code>class</code> as tags for the elemental
composition of the microbial communities. The <code>phylum</code> column
represents a major taxonomic rank, giving us a broad overview of the
microbial distribution. The <code>class</code> column provides more
detailed information, allowing us to delve deeper into the specific
types of microbes present.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathchee</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gAnnotation.html">gAnnotation</a></span><span class="op">(</span><span class="va">annotations</span>,<span class="st">"phylum"</span>,<span class="st">"class"</span><span class="op">)</span></span></code></pre></div>
<p>After pre-processing, here is what the two datasets that we will put
into the dashboard look like:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># data</span></span>
<span><span class="va">cheesedata</span><span class="op">[</span><span class="fl">112</span><span class="op">:</span><span class="fl">115</span>,<span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 11</span></span></span>
<span><span class="co">#&gt;   ID      A_1    A_2    A_3    A_4    A_5    C_1    C_3    C_4    C_5 type </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 112   0.943 -<span style="color: #BB0000;">0.471</span> -<span style="color: #BB0000;">0.471</span> -<span style="color: #BB0000;">0.471</span> -<span style="color: #BB0000;">0.471</span>  2.36  -<span style="color: #BB0000;">0.471</span> -<span style="color: #BB0000;">0.471</span> -<span style="color: #BB0000;">0.471</span> Other</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 113   0.786 -<span style="color: #BB0000;">0.124</span>  0.126 -<span style="color: #BB0000;">0.623</span> -<span style="color: #BB0000;">0.637</span>  2.33  -<span style="color: #BB0000;">0.667</span> -<span style="color: #BB0000;">0.593</span> -<span style="color: #BB0000;">0.593</span> Other</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 114   1.22  -<span style="color: #BB0000;">1.09</span>  -<span style="color: #BB0000;">0.430</span> -<span style="color: #BB0000;">0.829</span> -<span style="color: #BB0000;">0.170</span>  1.43  -<span style="color: #BB0000;">1.30</span>   0.455  0.715 Other</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 115   2.67  -<span style="color: #BB0000;">0.333</span> -<span style="color: #BB0000;">0.333</span> -<span style="color: #BB0000;">0.333</span> -<span style="color: #BB0000;">0.333</span> -<span style="color: #BB0000;">0.333</span> -<span style="color: #BB0000;">0.333</span> -<span style="color: #BB0000;">0.333</span> -<span style="color: #BB0000;">0.333</span> Other</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># annotation</span></span>
<span><span class="va">pathchee</span><span class="op">[</span><span class="fl">112</span><span class="op">:</span><span class="fl">115</span>,<span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 9</span></span></span>
<span><span class="co">#&gt;   ID    GO_ID KEGG_ID          domain Pathway taxonomic.scope order family genus</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 145   -     ko:K00004        Bacte… Pseudo… Unknown         <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 147   -     ko:K00004        Bacte… Actino… Actinomycetes   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 148   -     ko:K00004        Bacte… Bacill… Bacilli         Baci… Bacil… Lysi…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 149   -     ko:K00004,ko:K0… Bacte… Actino… Actinomycetes   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span></span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="cluster-input">Cluster Input<a class="anchor" aria-label="anchor" href="#cluster-input"></a>
</h3>
<p>In this section, we generate clusters from the first dataset above
using the <code>gClusters</code> function. This function takes the
cheese dataset (<code>cheesedata</code>) as input and generates clusters
based on specified parameters. Here, we set the number of clusters to 10
(<code>ncluster = 10</code>) and specify the maximum number of clusters
to consider for determining the optimal number of clusters
(<code>elbow.max=15</code>).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cluschee</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gClusters.html">gClusters</a></span><span class="op">(</span><span class="va">cheesedata</span>,ncluster <span class="op">=</span> <span class="fl">10</span>,elbow.max<span class="op">=</span><span class="fl">15</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="network-input">network input<a class="anchor" aria-label="anchor" href="#network-input"></a>
</h3>
<p>After generating the clusters for major patterns, we proceed to
obtain network results from the clusters. Taking the cluster centroids
as the nodes, the prediction process for edges is divided into
individual regression tasks, where each cluster centroid is
independentally predicted from the expression of all the other cluster
centroids, using random forests. We pick the top 3 related predictors
for each cluster centroid and save the network output for future use.
This is achieved using the <code><a href="../reference/gNetwork.html">gNetwork()</a></code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">networkchee</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gNetwork.html">gNetwork</a></span><span class="op">(</span><span class="va">cluschee</span>,ntop <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p>To gain more insight into the network results, we can visualize the
details using the <code><a href="../reference/gNetwork_view.html">gNetwork_view()</a></code> function, as shown
below.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/gNetwork_view.html">gNetwork_view</a></span><span class="op">(</span><span class="va">networkchee</span><span class="op">)</span></span></code></pre></div>
<p><img src="cheese_files/figure-html/unnamed-chunk-8-1.png" width="480"></p>
</div>
</div>
<div class="section level2">
<h2 id="run-dashboard">Run Dashboard<a class="anchor" aria-label="anchor" href="#run-dashboard"></a>
</h2>
<p>Once the clusters and network results are obtained, we can proceed to
run the dashboard. This involves declaring the annotations and then
executing the dashboard using the <code><a href="../reference/gDashboard.html">gDashboard()</a></code> function.
Here, we pass the cheese dataset (cheesedata), cluster results
(cluschee), network results (networkchee), and specify column names and
types for annotation identifiers.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/gDashboard.html">gDashboard</a></span><span class="op">(</span><span class="va">cheesedata</span>,</span>
<span>           <span class="va">cluschee</span>,</span>
<span>           <span class="va">pathchee</span>,</span>
<span>           <span class="va">networkchee</span>,</span>
<span>           id_colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO_ID"</span>,<span class="st">"KEGG_ID"</span><span class="op">)</span>,</span>
<span>           id_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO"</span>,<span class="st">"KEGG"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kaiyan Ma.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
